aws/data_source_aws_ec2_instance_type.go

```diff
@@ -0,0 +1,440 @@
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_gpu_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"fpgas": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_fpga_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"supported_placement_strategies": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"accelerators": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"hibernation_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"burstable_performance_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"dedicated_hosts_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"auto_recovery_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEc2InstanceTypeRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	params := &ec2.DescribeInstanceTypesInput{}
+
+	if v, ok := d.GetOk("instance_type"); ok {
+		params.InstanceTypes = []*string{aws.String(v.(string))}
+	}
+	log.Printf("[DEBUG] Reading instances types: %s", params)
+
+	resp, err := conn.DescribeInstanceTypes(params)
+	if err != nil {
+		return err
+	}
+	if len(resp.InstanceTypes) == 0 {
+		return fmt.Errorf("no EC2 Instance Type found matching criteria; try different search")
+	}
+	v := resp.InstanceTypes[0]
+	if err := d.Set("instance_type", aws.StringValue(v.InstanceType)); err != nil {
+		return fmt.Errorf("error setting instances_types: %s", err)
+	}
```

Comment:
For many of these `Set()` calls, this error handling is excessive. Also, `Set()` handles AWS types. Play around with it to make sure it is doing what you intended.

```suggestion
	d.Set("instance_type", v.InstanceType)
```